libtenzir/include/tenzir/actors.hpp

```diff
@@ -202,6 +202,8 @@ using catalog_actor = typed_actor_fwd<
   auto(atom::merge, std::vector<partition_synopsis_pair>)->caf::result<atom::ok>,
   // Get *ALL* partition synopses stored in the catalog.
   auto(atom::get)->caf::result<std::vector<partition_synopsis_pair>>,
+  // Get *ALL* partition synopses stored in the catalog for a given predicate.
+  auto(atom::get, expression)->caf::result<std::vector<partition_synopsis_pair>>,
```

Comment:
I'm not a fan of giving out pointers to internal memory from the catalog here, but it seems to be nothing new. Could you please add a comment to warn the next coder to never write to the structure returned by this handler and the one above?

Reply:
Done.